ql/timeseries.hpp

```diff
@@ -110,15 +110,34 @@ namespace QuantLib {
         typedef typename Container::const_iterator const_iterator;
         typedef typename const_iterator::iterator_category iterator_category;
 
+#if __cplusplus < 201703L
+        template <class ...>
+        using void_t = void;
+#else
+        using std::void_t;
+#endif
+
+        template <class container, typename = void>
+        struct enable_reverse final : std::false_type {
+        };
+
+        template <class container>
+        struct enable_reverse<container, void_t<typename container::const_reverse_iterator>> final : std::true_type {
+        };
+
+        template <class container, typename = void>
+        struct reverse {
+        };
+
```

Comment:
Before going into this, I would check if we really need this class to have a `Container` template argument or if we can just choose one and be done with it.